### PR TITLE
Add decision log for waiters

### PR DESCRIPTION
**Pull request by zoewangg**
## Description
Add decision log for waiters

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the **CONTRIBUTING** document
- [ ] Local run of `mvn install` succeeds
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have read the **README** document
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: docs/design/core/waiters/DecisionLog.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+when service waiters are not available.
+4. We should keep the union type `ResponseOrException` because it clearly indicates response and exception are mutually exclusive and only one is present. We should keep the naming `ResponseOrException` because it's intuitive and descriptive. 
+5. We should rename `WaiterResponse#responseOrExecption` -> `WaiterResponse#matched` because it has the most votes compared with other options: `matchedResponse`, `matchedResult`, `matchedValue`
+6. We should rename `{Service}Waiter.Builder#executorService` -> `{Service}Waiter.Builder#scheduledExecutorService` to be more clear it's a `ScheduledExecutorService`
+7. We should rename `Waiter.Builder#pollingStrategy` -> `Waiter.Builder#waiterOverrideConfiguration` so that we can easily add more configurations if needed. All configs under `WaiterOverrideConfiguration` should have default values to be consistent
+with `ClientOverrideConfiguration`
+8. We should create an overload method which takes a `WaiterOverrideConfiguration` parameter to allow request-level config override for every waiter operation, eg: `DynamodbWaiter#waitUntilTableExists(DescribeTableRequest, WaiterOverrideConfiguration)`
```
Comment (millems) [inline docs/design/core/waiters/DecisionLog.md:37]: This should usually follow the "Question? Decision. Justification." format, even if we decide not to change anything.

Comment (millems) [reply docs/design/core/waiters/DecisionLog.md:37]: Might be good to have an official scribe for the future meetings, so they can capture questions, even if they don't have action items. It seems like these are just the questions we decided to act on.

Comment (cenedhryn) [reply docs/design/core/waiters/DecisionLog.md:37]: +1

**Review by cenedhryn — APPROVED** [review]

File: docs/design/core/waiters/DecisionLog.md (hunk 1 above)
Comment (zoewangg) [reply docs/design/core/waiters/DecisionLog.md:37]: I'll update it to question format

**Pull request merged**